script.js

```diff
@@ -1,12 +1,440 @@
+    showImageSummaryInfoContainerHtml.classList.add(
+      "show-image-summary-info-container"
+    );
+
+    // Get the show name
+    const showName = show.name;
+    const showNameTextHtml = document.createElement("p");
+    const showNameContainerHtml = document.createElement("div");
+    showNameTextHtml.classList.add("show-name-text");
+    showNameContainerHtml.classList.add("show-name-container");
+    showNameTextHtml.innerHTML = showName;
+
+    // Get the show image
+
+    const showImageSrc = show.image?.medium;
+    const showImageHtml = document.createElement("img");
+    const showImageContainerHtml = document.createElement("div");
+    showImageHtml.classList.add("medium-img");
+    showImageContainerHtml.classList.add("show-image-container");
+    showImageHtml.src = showImageSrc;
+
+    // Get the show summary
+
+    const showSummary = show.summary;
+    const showSummaryTextHtml = document.createElement("p");
+    const showSummaryContainerHtml = document.createElement("div");
+    showSummaryTextHtml.classList.add("show-summary-text");
+    showSummaryContainerHtml.classList.add("show-summary-container");
+    showSummaryTextHtml.innerHTML = showSummary;
+
+    // Create the show basic info container elements
+
+    const showBasicInfoContainerHtml = document.createElement("div");
+    showBasicInfoContainerHtml.classList.add("show-info-container");
+
+    // accessing the elements from the data
+    const showRating = show.rating.average;
+    const showGenre = show.genres;
+    const showStatus = show.status;
+    const showRuntime = show.runtime;
+
+    // creating a rating element and its value inside showBasicInfoContainerHtml
+
+    const showRatingDescriptionHtml = document.createElement("span");
+    const showRatingValueHtml = document.createElement("span");
+    const showRatingContainer = document.createElement("p");
+    showRatingValueHtml.classList.add("show-rating-value");
+    showRatingDescriptionHtml.classList.add("show-rating-description");
+    showRatingContainer.classList.add("show-rating-container");
+    showRatingDescriptionHtml.innerText = "Rated: ";
+    showRatingValueHtml.innerHTML = `${showRating}`;
+
+    // creating a genre element and its value inside showBasicInfoContainerHtml
+
+    const showGenreDescriptionHtml = document.createElement("span");
+    const showGenreValueHtml = document.createElement("span");
+    const showGenreContainer = document.createElement("p");
+    showGenreValueHtml.classList.add("show-genre-value");
+    showGenreDescriptionHtml.classList.add("show-genre-description");
+    showGenreContainer.classList.add("show-genre-container");
+    showGenreDescriptionHtml.innerText = "Genres: ";
+    showGenreValueHtml.innerHTML = `${showGenre}`.split(",").join(", ");
+
+    // creating a status element and its value inside showBasicInfoContainerHtml
+
+    const showStatusValueHtml = document.createElement("span");
```

Comment:
I'm seeing a pattern of a lot of "document.createElement ; element.classList.add ; element.innerHTML =" (and also then `appendChild`) - is there a useful function to extract here which does these three things? It may make it more clear to see what's going on.

Reply:
You are right about the pattern. Will it be something like createElementAddClassAppend function? 
I'm not sure about the innerHTML. Sometimes we use .innerHTML and sometimes we use .src.